Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -996,6 +383,165 @@ bool CGImageIsMask(CGImageRef image) {
  @Notes
 */
 CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* components) {
+    RETURN_NULL_IF(!image);
     UNIMPLEMENTED();
     return StubReturn();
-}
\ No newline at end of file
+}
+
+#pragma region WIC_HELPERS
+
+CGImageRef _CGImageGetImageFromData(void* data, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_NULL, data, length);
+}
+
+CGImageRef _CGImageLoadGIF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatGif, bytes, length);
+}
+
+CGImageRef _CGImageLoadBMP(void* bytes, size_t length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatBmp, bytes, length);
+}
+
+CGImageRef _CGImageLoadTIFF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatTiff, bytes, length);
+}
+
+CGImageRef _CGImageLoadPNG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatPng, bytes, length);
+}
+
+CGImageRef _CGImageLoadJPEG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatJpeg, bytes, length);
+}
+
+CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void* bytes, int length) {
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+
+    ComPtr<IWICBitmapDecoder> pDecoder;
+    ComPtr<IWICStream> spStream;
+    RETURN_NULL_IF_FAILED(imageFactory->CreateStream(&spStream));
+    RETURN_NULL_IF_FAILED(spStream->InitializeFromMemory(static_cast<unsigned char*>(bytes), length));
+
+    if (!IsEqualGUID(decoderCls, GUID_NULL)) {
+        RETURN_NULL_IF_FAILED(imageFactory->CreateDecoder(decoderCls, nullptr, &pDecoder));
+        RETURN_NULL_IF_FAILED(pDecoder->Initialize(spStream.Get(), WICDecodeMetadataCacheOnLoad));
+    } else {
+        RETURN_NULL_IF_FAILED(imageFactory->CreateDecoderFromStream(spStream.Get(), nullptr, WICDecodeMetadataCacheOnDemand, &pDecoder));
```

Comment:
I still wonder what happens if you just fold these together and pass the decoder vendor